jwst/tweakreg/tests/test_tweakreg.py

```diff
@@ -21,6 +28,42 @@ def dummy_source_catalog():
     return catalog
 
 
+@pytest.mark.parametrize("inplace", [True, False])
+def test_rename_catalog_columns(dummy_source_catalog, inplace):
+    renamed_catalog = tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+
+    # if testing inplace, check the input catalog
+    if inplace:
+        catalog = dummy_source_catalog
+    else:
+        catalog = renamed_catalog
+
+    assert 'xcentroid' not in catalog.colnames
+    assert 'ycentroid' not in catalog.colnames
+    assert 'y' in catalog.colnames
+    assert 'y' in catalog.colnames
+
+
+@pytest.mark.parametrize("missing", ["x", "y", "xcentroid", "ycentroid"])
+def test_rename_catalog_columns_invalid(dummy_source_catalog, missing):
+    # if the column we want to remove is not in the table, first run
+    # rename to rename columns this should add the column we want to remove
+    if missing not in dummy_source_catalog.colnames:
+        tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+    dummy_source_catalog.remove_column(missing)
+    with pytest.raises(ValueError, match="catalogs must contain"):
+        tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+
+
+def test_rename_catalog_columns_inplace(dummy_source_catalog):
+    catalog = tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+    assert 'xcentroid' not in catalog.colnames
+    assert 'ycentroid' not in catalog.colnames
+    assert 'y' in catalog.colnames
+    assert 'y' in catalog.colnames
+
+
+@pytest.mark.skip(reason="test will need to be refactored as _is_wcs_correction_small is gone in favor of comparing skycoords")
```

Comment:
It's similar however the existing `_is_wcs_correction_small` compares 2 `wcs` objects. As `align_wcs` modifies the input `wcs` the original/input `wcs` would need to be copied or re-accessed (this is one of the places where the `corrector` used the reference to the input model removed in this PR). To avoid the copy (and avoid re-accessing the input model) this PR pre-computes the skycoord and adds it to the `corrector` metadata. This changes the input for `_is_wcs_correction_small` from 2 `wcs` objects to 1 `wcs` and a `skycoord`.

I originally removed `_is_wcs_correction_small` and added the check to the step (and forgot to remove the test). However, to preserve the test, I updated this PR modifying `_is_wcs_correction_small` and the corresponding test in: https://github.com/spacetelescope/jwst/pull/8424/commits/204f764bd7e3113fc50ea3547baacb844670eba0

Thanks for the comment! An updated test is better than a skipped one :)